### PR TITLE
test: add some test case

### PR DESCRIPTION
**Pull request by Gala-R**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,7 +50,7 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 test: manifests generate fmt vet envtest ## Run tests.
-	go test -coverpkg=./... -coverprofile=cover.out --timeout=20m ./...
+	go test $(go list ./...  | grep -v ./emqx-operator/apis/apps/v1beta3/zz_generated.deepcopy.go) -coverprofile=cover.out --timeout=20m ./...
```
Comment (Rory-Z) [inline Makefile:53]: I don't think that's a good way to go, we'll have many apiVersions

File: controllers/apps/v1beta3/emqxplugin_controller.go [hunk 2]
```diff
@@ -117,31 +117,21 @@ func (r *EmqxPluginReconciler) Reconcile(ctx context.Context, req ctrl.Request)
...
-				ready = true
-			}
-		}
-		if !ready {
-			break
-		}
-
-		equalPluginConfig, err := r.checkPluginConfig(instance, emqx)
-		if err != nil {
+	ready, emqx := findReadyEmqx(emqxList)
```
Comment (Rory-Z) [inline controllers/apps/v1beta3/emqxplugin_controller.go:120]: Why didn't use `handler. findReadyEmqx` ?

File: controllers/apps/v1beta3/suite_test.go [hunk 3]
```diff
@@ -149,6 +149,21 @@ var _ = BeforeSuite(func() {
...
+	broker.SetNodeName("")
+	broker.SetNodeSelector(nil)
+	broker.SetImagePullSecrets(nil)
+	broker.SetSecurityContext(nil)
+	broker.SetInitContainers(nil)
+	broker.SetAffinity(nil)
+	broker.SetLivenessProbe(nil)
+	broker.SetReadinessProbe(nil)
+	broker.SetStartupProbe(nil)
+	broker.SetReplicas(int32Ptr(3))
```
Comment (Rory-Z) [inline controllers/apps/v1beta3/suite_test.go:166]: Delete this, these do not make sense

File: controllers/apps/v1beta3/suite_test.go [hunk 4]
```diff
@@ -165,8 +180,22 @@ var _ = BeforeSuite(func() {
...
+	enterprise.SetNodeName("")
+	enterprise.SetNodeSelector(nil)
+	enterprise.SetImagePullSecrets(nil)
+	enterprise.SetSecurityContext(nil)
+	enterprise.SetInitContainers(nil)
+	enterprise.SetAffinity(nil)
+	enterprise.SetLivenessProbe(nil)
+	enterprise.SetReadinessProbe(nil)
+	enterprise.SetStartupProbe(nil)
+	enterprise.SetReplicas(int32Ptr(3))
```
Comment (Rory-Z) [inline controllers/apps/v1beta3/suite_test.go:197]: Delete this, these do not make sense

**Review by Rory-Z — CHANGES_REQUESTED** [review]

File: controllers/apps/v1beta3/suite_test.go [hunk 5]
```diff
@@ -253,6 +255,16 @@ func cleanAll() {
...
 	Expect(k8sClient.Delete(context.Background(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: enterprise.GetNamespace()}})).Should(Succeed())
+
+	Eventually(func() bool {
+		ns, _ := clientset.CoreV1().Namespaces().Get(context.Background(), broker.GetNamespace(), metav1.GetOptions{})
+		return len(ns.Name) == 0
+	}, time.Second*120, time.Second*10).Should(BeTrue())
+
+	Eventually(func() bool {
+		ns, _ := clientset.CoreV1().Namespaces().Get(context.Background(), enterprise.GetNamespace(), metav1.GetOptions{})
+		return len(ns.Name) == 0
```
Comment (Rory-Z) [inline controllers/apps/v1beta3/suite_test.go:266]:
```suggestion
		err := k8sClient.Get(context.Background(), client.ObjectKey{Name: "kube-system"}, &corev1.Namespace{})
		k8sError.IsNotFound(err)
```

File: controllers/apps/v1beta3/emqxplugin_controller.go (hunk 2 above)
Comment (Gala-R) [reply controllers/apps/v1beta3/emqxplugin_controller.go:120]: the parameters are different

File: controllers/apps/v1beta3/suite_test.go (hunk 4 above)
Comment (Gala-R) [reply controllers/apps/v1beta3/suite_test.go]: Done

File: controllers/apps/v1beta3/suite_test.go (hunk 3 above)
Comment (Gala-R) [reply controllers/apps/v1beta3/suite_test.go]: Done

File: Makefile (hunk 1 above)
Comment (Gala-R) [reply Makefile]: already rolled back

File: controllers/apps/v1beta3/suite_test.go (hunk 5 above)
Comment (Gala-R) [reply controllers/apps/v1beta3/suite_test.go]: Done

**Review by Rory-Z — APPROVED** [review]

**Pull request merged**